ibazel/ibazel_test.go

```diff
@@ -35,17 +35,39 @@ func assertEqual(t *testing.T, want, got interface{}, msg string) {
 	}
 }
 
-func assertKilled(t *testing.T, cmd *exec.Cmd) {
-	if err := cmd.Wait(); err != nil {
-		if cmd.ProcessState.Success() {
-			t.Errorf("Subprocess terminated from \"natural\" causes, which means the job ran for 5 sec then existed. The Run method should have killed it before then.")
-		}
-		if cmd.ProcessState == nil {
-			t.Errorf("Killable subprocess was never started. State: %v, Err: %v", cmd.ProcessState, err)
-		}
+type mockCommand struct {
+	b      bazel.Bazel
+	target string
+	args   []string
+
+	notifiedOfChanges bool
+	started           bool
+	terminated        bool
+}
+
+func (m *mockCommand) Start() {
+	m.started = true
+}
+func (m *mockCommand) NotifyOfChanges() {
+	m.notifiedOfChanges = true
+}
+func (m *mockCommand) Terminate() {
+	if !m.started {
+		panic("Terminated before starting")
+	}
+	m.terminated = true
+}
+func (m *mockCommand) assertTerminated(t *testing.T) {
+	if !m.terminated {
+		t.Errorf("Mock command wasn't terminated")
+		debug.PrintStack()
 	}
```

Comment:
I love this :D. You are much less lazy than me Andrew

Reply:
Much less lazy :wink: